src/frame/sdk/SearchApi.ts

```diff
@@ -0,0 +1,90 @@
+    const salt = randomBytes.slice(0, CryptoConstants.SALT_LENGTH);
+    const randomHexDocId = encodeBytesAsHex(randomBytes.slice(CryptoConstants.SALT_LENGTH));
+    return encrypt(randomHexDocId, salt, [], [groupId], false)
+        .map((unmanagedDoc) => ({
+            searchIndexEncryptedSalt: unmanagedDoc.document,
+            searchIndexEdeks: unmanagedDoc.edeks,
+        }))
+        .errorMap((e) => new SDKError(e.rawError, ErrorCodes.SEARCH_CREATE_INDEX_FAILURE));
+};
+
+/**
+ * Initialize an existing blind index search. Decrypts the provided unmanaged document salt and stores the result in memory
+ * keyed by the document ID of the blind index.
+ */
+export const initializeBlindSearchIndex = (encryptedSalt: Uint8Array, searchIndexEdeks: Uint8Array): Future<SDKError, string> => {
+    const searchIndexId = encodeBytesAsHex((window.msCrypto || window.crypto).getRandomValues(new Uint8Array(8)));
+    return decryptWithProvidedEdeks(encryptedSalt, searchIndexEdeks)
+        .map((index) => {
+            searchIndexCache[searchIndexId] = index.data;
+            return searchIndexId;
+        })
+        .errorMap((e) => new SDKError(e.rawError, ErrorCodes.SEARCH_INIT_INDEX_FAILURE));
+};
+
+/**
+ * Tokenize the provided search data using the provided searchIndexId's decrypted salt and optional partition ID.
+ */
+export const tokenizeData = (searchIndexId: string, data: string, partitionId?: string): Future<SDKError, Uint32Array> => {
+    if (!searchIndexCache[searchIndexId]) {
+        return Future.reject(new SDKError(new Error("This search index has not yet been initialized."), ErrorCodes.SEARCH_TOKENIZE_DATA_FAILURE));
+    }
+    const payload: WMT.SearchTokenizeDataRequest = {
+        type: "SEARCH_TOKENIZE_DATA",
+        message: {
+            value: data,
+            partitionId,
+            salt: searchIndexCache[searchIndexId],
+        },
+    };
+    return WorkerMediator.sendMessage<WMT.SearchTokenizeStringResponse>(payload).map(({message}) => message);
+};
+
+/**
+ * Tokenize the provided search query using the provided searchIndexId's decrypted salt and optional partition ID.
+ */
+export const tokenizeQuery = (searchIndexId: string, query: string, partitionId?: string): Future<SDKError, Uint32Array> => {
+    if (!searchIndexCache[searchIndexId]) {
+        return Future.reject(new SDKError(new Error("This search index has not yet been initialized."), ErrorCodes.SEARCH_TOKENIZE_QUERY_FAILURE));
+    }
+    const payload: WMT.SearchTokenizeQueryRequest = {
+        type: "SEARCH_TOKENIZE_QUERY",
+        message: {
+            value: query,
+            partitionId,
+            salt: searchIndexCache[searchIndexId],
+        },
+    };
+    return WorkerMediator.sendMessage<WMT.SearchTokenizeStringResponse>(payload).map(({message}) => message);
+};
+
+/**
+ * Pass provided string to WebWorker for transliteration
+ */
+export const transliterateString = (string: string) => {
```

Comment:
Why/when is this needed?

Reply:
It's hopefully described in the ticket that [added this method](https://github.com/IronCoreLabs/search-helpers/issues/9) to `search-helpers`, but if that doesn't describe it enough, let me know.